lambda/translcore.ml

```diff
@@ -31,7 +31,7 @@ type error =
 
 exception Error of Location.t * error
 
-let use_dup_for_constant_arrays_bigger_than = 4
+let use_dup_for_constant_mutable_arrays_bigger_than = 4
```

Comment:
I'll read the changes to this module (hopefully tomorrow).
We should try to arrange things so that immutable arrays at the source level (except for ones constructed with comprehensions) turn into immutable arrays in Lambda -- this means that e.g. projection from array literals can be evaluated by the middle-end.  I'm unsure if this is happening at present.

Reply:
(and also it means that things like nested immutable array literals can be statically allocated)

Reply:
@goldfirere @antalsz I've read these `Translcore` changes, I think they're ok - but we should test carefully, this stuff is tricky.